src/pages/index.js

```diff
@@ -30,55 +36,198 @@ const addCardPopup = new PopupWithForm(
   "#profile__add-card-modal",
   handleAddCardSubmit
 );
+const editAvatarPopup = new PopupWithForm(
+  "#profile__avatar-modal",
+  handleAvatarSubmit
+);
+profileEditPopup.setEventListeners();
+addCardPopup.setEventListeners();
+editAvatarPopup.setEventListeners();
+
 const previewImagePopup = new PopupWithImage("#card_modal");
+const trashConfirmPopup = new PopupWithConfirm("#trashcan-modal", handleDelete);
 
 // Section
-const cardSection = new Section(
-  { items: initialCards, renderer },
-  ".cards__list"
-);
-cardSection.renderItems();
+const cardSection = new Section({ items: [], renderer }, ".cards__list");
 
-// Form validators
+// Validators
+const profileEditForm = document.querySelector("#profileEditForm");
+const addCardForm = document.querySelector("#addcard__form");
 const profileEditFormValidator = new FormValidator(config, profileEditForm);
 const addCardFormValidator = new FormValidator(config, addCardForm);
 profileEditFormValidator.enableValidation();
 addCardFormValidator.enableValidation();
 
-// Event Listeners
-profileEditBtn.addEventListener("click", () => {
-  const userData = userInfo.getUserInfo();
-  profileTitleInput.value = userData.name;
-  profileDescriptionInput.value = userData.about;
-  profileEditPopup.open();
-});
-
-addCardButton.addEventListener("click", () => addCardPopup.open());
+// Constants
+const profileEditBtn = document.querySelector("#profile__edit-button");
+const addCardButton = document.querySelector(".profile__add-button");
+const profileTitleInput = document.querySelector("#profile__name-input");
+const profileDescriptionInput = document.querySelector(
+  "#profile__subheading-input"
+);
+const trashModalSubmitBtn = document.querySelector(".modal__button-trash");
+const editProfileImage = document.querySelector(".profile__avatar-edit");
 
-// Event handlers
+// Functions
 function handleProfileEditSubmit(profileData) {
-  const { modal__input_type_name: name, modal__input_type_description: about } =
-    profileData;
-  userInfo.setUserInfo(name, about);
-  profileEditPopup.close();
+  const name = profileData.modal__input_type_name;
+  const about = profileData.modal__input_type_description;
+
+  profileEditPopup.renderLoading(true);
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```